### PR TITLE
qa/rgw: enable valgrind in rgw/notifications suite

### PR DESCRIPTION
**Pull request by cbodley**
to help debug crashes like https://tracker.ceph.com/issues/65337

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: qa/suites/rgw/notifications/valgrind.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+overrides:
+  install:
+    ceph:
+      #debuginfo: true
+  rgw:
+    client.0:
+      valgrind: [--tool=memcheck, --max-threads=1024] # http://tracker.ceph.com/issues/25214
+    client.1:
```
Comment (yuvalif) [inline qa/suites/rgw/notifications/valgrind.yaml:8]: i think that only the "others" task is using 2 RGWs. amqp and kafka have only "client.0"

**Comment by yuvalif** [comment]
can you please add teuthology run?

**Comment by cbodley** [comment]
i did two runs of the notifications suite using a [suite-branch](https://github.com/cbodley/ceph/commits/wip-qa-rgw-65337) that includes this and your fixes from https://github.com/ceph/ceph/pull/55663:

https://pulpito.ceph.com/cbodley-2024-04-09_17:48:09-rgw:notifications-wip-yuval-63909-distro-default-smithi/
https://pulpito.ceph.com/cbodley-2024-04-09_17:47:56-rgw:notifications-wip-yuval-63909-distro-default-smithi/

**Comment by yuvalif** [comment]
> i did two runs of the notifications suite using a [suite-branch](https://github.com/cbodley/ceph/commits/wip-qa-rgw-65337) that includes this and your fixes from #55663:
> 
> https://pulpito.ceph.com/cbodley-2024-04-09_17:48:09-rgw:notifications-wip-yuval-63909-distro-default-smithi/ 
> https://pulpito.ceph.com/cbodley-2024-04-09_17:47:56-rgw:notifications-wip-yuval-63909-distro-default-smithi/
* both amqp and kafka crashed here:
```
2024-04-09T18:15:27.803 INFO:tasks.rgw.client.1.smithi094.stdout:*** Caught signal (Segmentation fault) **
2024-04-09T18:15:27.803 INFO:tasks.rgw.client.1.smithi094.stdout: in thread a0430c0 thread_name:memcheck-amd64-
2024-04-09T18:15:27.856 INFO:tasks.rgw.client.1.smithi094.stdout: ceph version 19.0.0-2635-gdded6fa8 (dded6fa8fb3155a1d6c5dc77b4d3ab21f6c18cce) squid (dev)
2024-04-09T18:15:27.856 INFO:tasks.rgw.client.1.smithi094.stdout: 1: /lib64/libc.so.6(+0x54db0) [0x797bdb0]
2024-04-09T18:15:27.856 INFO:tasks.rgw.client.1.smithi094.stdout: 2: radosgw(+0x7bdeec) [0x8c5eec]
2024-04-09T18:15:27.857 INFO:tasks.rgw.client.1.smithi094.stdout: 3: radosgw(+0x2e8532) [0x3f0532]
2024-04-09T18:15:27.857 INFO:tasks.rgw.client.1.smithi094.stdout: 4: (void boost::context::detail::context_entry<boost::context::detail::record<boost::context::continuation, boost::context::basic_protected_fixedsize_stack<boost::context::stack_traits>, spawn::detail::spawn_helper<boost::asio::executor_binder<void (*)(), boost::asio::any_io_executor>, rgw::notify::Manager::process_queue(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, spawn::basic_yield_context<boost::asio::executor_binder<void (*)(), boost::asio::any_io_executor> >)::{lambda(spawn::basic_yield_context<boost::asio::executor_binder<void (*)(), boost::asio::any_io_executor> >)#7}, boost::context::basic_protected_fixedsize_stack<boost::context::stack_traits> >::operator()()::{lambda(boost::context::continuation&&)#1}> >(boost::context::detail::transfer_t)+0x55) [0x8d1e55]
2024-04-09T18:15:27.857 INFO:tasks.rgw.client.1.smithi094.stdout: 5: make_fcontext()
```
none of these should be doing realm reload (except for the initial setup).

* the http/basic/v2 migration test is crashing with the same backtrace mid migration (as we are "stuck" in migrating state)

**Comment by yuvalif** [comment]
@cbodley why not adding valgrind to these tests?

**Comment by yuvalif** [comment]
@cbodley ping

**Comment by cbodley** [comment]
@yuvalif as you mentioned in https://github.com/ceph/ceph/pull/57537#issuecomment-2200665477, testing uncovered valgrind issues. would you prefer to merge this anyway, or wait until we've resolved them?

**Comment by cbodley** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
> @yuvalif as you mentioned in [#57537 (comment)](https://github.com/ceph/ceph/pull/57537#issuecomment-2200665477), testing uncovered valgrind issues. would you prefer to merge this anyway, or wait until we've resolved them?



> @yuvalif as you mentioned in [#57537 (comment)](https://github.com/ceph/ceph/pull/57537#issuecomment-2200665477), testing uncovered valgrind issues. would you prefer to merge this anyway, or wait until we've resolved them?

i think this is only happening when there are issues with the amqp server.
IMO, no need to wait for fixe on that.

**Comment by cbodley** [comment]
jenkins test make check

**Comment by smanjara** [comment]
jenkins test make check

**Review by smanjara — APPROVED** [review]

**Comment by smanjara** [comment]
jenkins test make check

**Pull request merged**